File: src/commands/music/skip.ts
import type { GuildMember } from "discord.js";
import { defineCommand, ICommandInteraction } from "../../lib/util/valeriyya.types";

export default defineCommand({
  data: {
    name: "skip",
    description: "Skip the current playing song.",
  },
  chat: async (int: ICommandInteraction) => {
    const subscription = int.client.subscription.get(int.guildId!);
    const member = int.member as GuildMember;

    if (!subscription) return "There is no music currently playing.";
    if (member.voice.channelId !== int.guild!.me?.voice.channelId)
      return {
        content: "You must be in the same voice channel as me to use this command! <3",
        ephemeral: true,
      };

    subscription.audioPlayer.stop();

    return `${int.user} has skipped the song`;
  },
});

File: src/commands/music/skip.rs
use crate::{Context, Error, utils::Valeriyya};

/// Skips the currently playing song.
#[poise::command(prefix_command, slash_command, category = "Music", aliases("s"))]
pub async fn skip(ctx: Context<'_>) -> Result<(), Error> {
    let guild_id = ctx.guild_id().unwrap();

    let channel_id = ctx
        .guild()
        .unwrap()
        .voice_states
        .get(&ctx.author().id)
        .and_then(|voice_state| voice_state.channel_id);

    let _connect_to = match channel_id {
        Some(channel) => channel,
        None => {
            ctx.send(Valeriyya::reply("You are not in a voice channel").ephemeral(true)).await;
            return Ok(());
        }
    };

    if let Some(handler_lock) = ctx.data().songbird.get(guild_id) {
        let handler = handler_lock.lock().await;
        let queue = handler.queue();
        match queue.is_empty() {
            false => {
                queue.skip();
            }
            true => {
                drop(handler);
                ctx.send(Valeriyya::reply("There is no songs in the queue!").ephemeral(true)).await;
            }
        };
    };

    Ok(())
}
